orttraining/orttraining/core/graph/gradient_schema_defs.cc

```diff
@@ -1300,32 +1326,40 @@ Return true if all elements are true and false otherwise.
       .SinceVersion(1)
       .SetSupportLevel(OpSchema::SupportType::EXPERIMENTAL)
       .SetDoc("Send data tensor to the specified destination.")
-      .Input(0, "InputSignal", "Input control signal.", "TBool")
-      .Input(1, "Data", "Tensor to send.", "T")
-      .Output(0, "OutputSignal", "Output control signal.", "TBool")
-      .Attr("src",
-            "Abstractive memory ID of Data's source.",
-            AttributeProto::INT)
-      .Attr("dst",
-            "Abstractive memory ID of Data's destination.",
-            AttributeProto::INT)
+      .Input(0, "InputSignal", "Input control signal. It must be a scalar.", "TBool")
+      .Input(1, "Remote", "Remote dst rank. It must be a scalar.", "TInt64")
+      .Input(2, "Data", "Tensors to send.", "V", OpSchema::Variadic, false)
+      .Output(0, "OutputSignal", "Output control signal. It must be a scalar.", "TBool")
       .Attr("tag", "The tag of the message carrying Data.",
             AttributeProto::INT)
-      .Attr("element_type", "Element type of the sent tensor.",
-            AttributeProto::INT)
+      .Attr("element_types", "Element types of the sent tensors.",
+            AttributeProto::INTS)
       .TypeConstraint(
-          "T",
-          {"tensor(float16)", "tensor(float)", "tensor(double)"},
-          "Constrain input types to float tensors.")
+          "TInt64",
+          {"tensor(int64)"},
+          "Constrain input type to 64-bit integer.")
       .TypeConstraint(
           "TBool",
           {"tensor(bool)"},
           "Constrain types to boolean tensors.")
+      .TypeConstraint("V", OpSchema::all_tensor_types(), "All Tensor types")
       .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
-        if (ctx.getNumInputs() != 2)
-          fail_shape_inference("Send must have two inputs.");
-        if (ctx.getNumOutputs() != 1)
+        if (ctx.getNumInputs() < 3) {
+          fail_shape_inference("Send must have at least three inputs.");
+        } else {
```

Comment:
>else [](start = 10, length = 4)

Check number of data input == size of element_types ?

Reply:
sure.